downloadertask/IspwDownloader.ts

```diff
@@ -214,41 +218,68 @@ export class IspwDownloader {
+    Operating system Windows/Linux
+    */
+    getCommand(cliArguments: any): string | undefined {
+        //Windows = 0,
+        //MacOS = 1,
+        //Linux = 2
+        let command;
+        switch (tl.getPlatform()) {
+            case 0:
+                command = require.resolve(cliArguments.get("windowsTopazWorkbenchCliHome") + '/SCMDownloaderCLI.bat');
+                break;
+            case 2:
+                command = require.resolve(cliArguments.get("linuxTopazWorkbenchCliHome") + '/SCMDownloaderCLI.sh');
+                break;
+            default:
+                tl.setResult(tl.TaskResult.Failed, "Unsupported operating system!! Currently it only supports Windows and Linux Operating System");
+        }
+        console.log("Inside command function" + command);
+        return command;
+    }
+
     /*  
     Function for downloading source from 
     ISPW Repository
     */
     downloadRepositorySource(cliArguments: any): void {
-        //Calling bat command
-        var bat = require.resolve(cliArguments.get("windowsTopazWorkbenchCliHome") + '/SCMDownloaderCLI.bat');
-        var ls = spawn(bat, ['-host', cliArguments.get("host"), '-port', cliArguments.get("port"), '-code', cliArguments.get("codePage"), '-id',
-            cliArguments.get("userId"), '-pass', cliArguments.get("password"), '-targetFolder', cliArguments.get("windowsSourceDownloadLocation"),
-            '-scm', 'ISPW', '-ispwServerConfig', cliArguments.get("runtimeConfig"),'-ispwServerStream', cliArguments.get("stream"), '-ispwServerApp', cliArguments.get("application"),
-            '-ispwServerLevel', cliArguments.get("repositoryLevel"), '-ispwDownloadAll', cliArguments.get("downloadUnchangedSource"),
-            '-ispwDownloadIncl', cliArguments.get("downloadIncludes"), '-ispwLevelOption', cliArguments.get("levelOption"),
-            '-ispwComponentType',cliArguments.get("componentTypes"),'-ispwFolderName',cliArguments.get("applicationRootFolderNames"),
-            '-ispwDownloadWithCompileOnly',cliArguments.get("downloadCompileOnly"), '-ispwFilterFiles',
-            cliArguments.get("ispwFilterFiles"),'-ispwFilterFolders',cliArguments.get("ispwFilterFolders")
-        ]);
-        console.log(ls);
-
-        ls.stdout.on('data', function (data) {
-            console.log('stdout: ' + data);
-        });
-
-        ls.stderr.on('data', function (data) {
-            console.log('stderr: ' + data);
-        });
-
-        ls.on('exit', function (code) {
-            console.log('child process exited with code ' + code);
-            if(code!=null && code!=0)
-            {
-                tl.setResult(tl.TaskResult.Failed, " An error may have occurred. Please see task logs or see the log file: "+cliArguments.get("windowsTopazWorkbenchCliHome")+"\\TopazBatchWkspc\\.metadata\\.log.");
-            }
-        });
-
-        
-
+        let command = this.getCommand(cliArguments);
+        console.log(command);
```

Comment:
is this console log required ?

Reply:
This was just for draft purpose, we can remove this